sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2796,7 +2796,23 @@ checkDevSupportDeviceRequirements(const device &Dev,
       Dims++;
     }
 
-    if (NDRDesc.Dims != 0 && NDRDesc.Dims != static_cast<size_t>(Dims))
+    size_t UserProvidedNumDims = 0;
+    if (WorkGroupNumDim) {
+      // We know the dimensions have been padded to 3, make sure that the pad
+      // value is always set to 1 and record the number of dimensions specified
+      // by the user.
+#ifdef NDEBUG
```

Comment:
Sorry, I just realized it was the wrong way around.

```suggestion
#ifndef NDEBUG
```